src/markets.rs

```diff
@@ -0,0 +1,150 @@
+mod kraken;
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum TradingPair {
+    BtcDai,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct Rate {
+    trading_pair: TradingPair,
+    position: Position,
+    rate: f64,
+    timestamp: DateTime<Utc>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum Position {
+    BUY,
+    SELL,
+}
+
+pub struct Ohlc {
+    high: f64,
+    low: f64,
+    vwap: f64,
+    timestamp: DateTime<Utc>,
+    trading_pair: TradingPair,
+}
+
+impl Ohlc {
+    fn to_rate(&self, position: Position) -> anyhow::Result<Rate> {
+        let rate = if self.vwap == 0.0 {
+            let precision = 0.000_000_000_000_001;
+            if (self.high - self.low).abs() > precision {
+                tracing::warn!("OHLC high and low value are not the same even though there were no trades recorded (vwap 0).")
```

Comment:
:question: Why do we care? And what can/should we do about it?

Reply:
We currently assume, that the API behaves in the way that `high` and `low` would always be the same unless there is a `vwap` value is set. But we cannot be 100% sure, because the documentation does not elaborate on this. 

Taking the `high` value should be save anyway - the log message is more to notice that the API actually behaves different than we expect. I can remove it if it is confusing. I don't think it will make a big difference.